tests/integration/encryption/commit_test.go

```diff
@@ -0,0 +1,359 @@
+				Doc: `{
+						"name":	"John",
+						"age":	21
+					}`,
+				IsEncrypted: true,
+			},
+			testUtils.CreateDoc{
+				Doc: `{
+						"name":	"Islam",
+						"age":	33
+					}`,
+			},
+			testUtils.UpdateDoc{
+				DocID: 0,
+				Doc: `{
+					"age": 22
+				}`,
+			},
+			testUtils.UpdateDoc{
+				DocID: 1,
+				Doc: `{
+					"age": 34
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						commits(fieldId: "1") {
+							delta
+							docID
+						}
+					}
+				`,
+				Results: []map[string]any{
+					{
+						"delta": encrypt(testUtils.CBORValue(22)),
+						"docID": johnDocID,
+					},
+					{
+						"delta": encrypt(testUtils.CBORValue(21)),
+						"docID": johnDocID,
+					},
+					{
+						"delta": testUtils.CBORValue(34),
+						"docID": islamDocID,
+					},
+					{
+						"delta": testUtils.CBORValue(33),
+						"docID": islamDocID,
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestDocEncryption_WithEncryptionOnCounterCRDT_ShouldStoreCommitsDeltaEncrypted(t *testing.T) {
+	const docID = "bae-d3cc98b4-38d5-5c50-85a3-d3045d44094e"
+
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+                    type Users {
+                        points: Int @crdt(type: "pcounter")
+                    }
+                `},
+			testUtils.CreateDoc{
+				Doc:         `{ "points": 5 }`,
+				IsEncrypted: true,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						commits {
+							cid
+							delta
+							docID
+						}
+					}
+				`,
+				Results: []map[string]any{
+					{
+						"cid":   "bafyreieb6owsoljj4vondkx35ngxmhliauwvphicz4edufcy7biexij7mu",
+						"delta": encrypt(testUtils.CBORValue(5)),
+						"docID": docID,
+					},
+					{
+						"cid":   "bafyreif2lejhvdja2rmo237lrwpj45usrm55h6gzr4ewl6gajq3cl4ppsi",
+						"delta": nil,
+						"docID": docID,
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestDocEncryption_UponUpdateOnCounterCRDT_ShouldEncryptedCommitDelta(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+                    type Users {
+                        points: Int @crdt(type: "pcounter")
+                    }
+                `},
+			testUtils.CreateDoc{
+				Doc:         `{ "points": 5 }`,
+				IsEncrypted: true,
+			},
+			testUtils.UpdateDoc{
+				Doc: `{
+					"points": 3
+				}`,
+			},
+			testUtils.Request{
+				Request: `
+					query {
+						commits(fieldId: "1") {
+							delta
+						}
+					}
+				`,
+				Results: []map[string]any{
+					{
+						"delta": encrypt(testUtils.CBORValue(3)),
+					},
+					{
+						"delta": encrypt(testUtils.CBORValue(5)),
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestDocEncryption_UponEncryptionSeveralDocs_ShouldStoreAllCommitsDeltaEncrypted(t *testing.T) {
+	const johnDocID = "bae-c9fb0fa4-1195-589c-aa54-e68333fb90b3"
```

Comment:
suggestion: I don't think we care about the docID value in these commit tests?  Can we perhaps use `testUtils.NewDocIndex(...)` instead?

Reply:
done